tests/aws/services/events/test_event_patterns.py

```diff
@@ -0,0 +1,160 @@
+
+
+def load_request_templates(directory_path: str) -> List[Tuple[dict, str]]:
+    json5_files = list_files_with_suffix(directory_path, ".json5")
+    return [load_request_template(file_path) for file_path in json5_files]
+
+
+def load_request_template(file_path: str) -> Tuple[dict, str]:
+    with open(file_path, "r") as df:
+        template = json5.load(df)
+    return template, Path(file_path).stem
+
+
+def list_files_with_suffix(directory_path: str, suffix: str) -> List[str]:
+    files = []
+    for root, _, filenames in os.walk(directory_path):
+        for filename in filenames:
+            if filename.endswith(suffix):
+                absolute_filepath = os.path.join(root, filename)
+                files.append(absolute_filepath)
+
+    return files
+
+
+request_template_tuples = load_request_templates(REQUEST_TEMPLATE_DIR)
+
+SKIP_LABELS = [
+    # Failing exception tests:
+    "arrays_empty_EXC",
+    "content_numeric_EXC",
+    "content_numeric_operatorcasing_EXC",
+    "content_numeric_syntax_EXC",
+    "content_wildcard_complex_EXC",
+    "int_nolist_EXC",
+    "operator_case_sensitive_EXC",
+    "string_nolist_EXC",
+    # Failing tests:
+    "complex_or",
+    "content_anything_but_ignorecase",
+    "content_anything_but_ignorecase_list",
+    "content_anything_suffix",
+    "content_exists_false",
+    "content_ignorecase",
+    "content_ignorecase_NEG",
+    "content_ip_address",
+    "content_numeric_and",
+    "content_prefix_ignorecase",
+    "content_suffix",
+    "content_suffix_ignorecase",
+    "content_wildcard_nonrepeating",
+    "content_wildcard_repeating",
+    "content_wildcard_simplified",
+    "dot_joining_event",
+    "dot_joining_pattern",
+    "nested_json_NEG",
+    "or-exists",
+    "or-exists-parent",
+]
+
+
+# TODO: extend these test cases based on the open source docs + tests: https://github.com/aws/event-ruler
+#  For example, "JSON Array Matching", "And and Or Relationship among fields with Ruler", rule validation,
+#  and exception handling.
+@pytest.mark.skipif(is_v2_provider(), reason="V2 provider does not support this feature yet")
+@pytest.mark.parametrize(
+    "request_template,label", request_template_tuples, ids=[t[1] for t in request_template_tuples]
+)
+@markers.aws.validated
+def test_test_event_pattern(aws_client, snapshot, request_template, label):
+    """This parametrized test handles three outcomes:
+    a) MATCH (default): The EventPattern matches the Event yielding true as result.
+    b) NO MATCH (_NEG suffix): The EventPattern does NOT match the Event yielding false as result.
+    c) EXCEPTION (_EXC suffix): The EventPattern is invalid and raises an exception.
+    """
+    if label in SKIP_LABELS and not is_aws_cloud():
+        pytest.skip("Not yet implemented")
+
+    event = request_template["Event"]
+    event_pattern = request_template["EventPattern"]
+
+    if label.endswith("_EXC"):
```

Comment:
What was your reasoning here for encoding this in the file name, instead of having another field in the json5 file?

Reply:
Good question. It kinda evolved because I wanted to have an explicit way to express the intent of the test. The surprise effect helps to write better test cases. It could as well be a field in the the json5 file though.

Not really a strong reason, some thoughts:

* The intention of the test is easy to see in the file explorer.
* It currently matches the [TestEventPattern](https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_TestEventPattern.html) API. Maybe, it would be easier to copy/paste and test in the AWS console.

Reply:
I initially built a small templating engine supporting variables but noticed we can simplify and don't care about any dynamic values 🤦  So the entire request "templating" part comes from there 😅